src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3234,18 +3235,32 @@ static size_t large_page_init_decide_size() {
     return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  if (!os::win32::is_windows_11_or_greater() && !os::win32::is_windows_server_2022_or_greater()) {
+#if defined(IA32)
```

Comment:
Do we need a warning that the EnableAllLargePageSizes flag will be ignored on this architecture?

Reply:
how will this warning help, do we have a usecase where the users in ia32 might try to use the EnableAllLargePageSizes flag ... please help me understand

Reply:
> how will this warning help, do we have a usecase where the users in ia32 might try to use the EnableAllLargePageSizes flag ... please help me understand

I'm not claiming there is a use case for IA32. Maybe the answer here is that we don't care. I'm simply observing that this flag is only checked on AMD64 - I'm not sure whether that is intentional because I assume the flag will be available on IA32 as well.

Reply:
@nitinsingh130519 , is LargePageSizeInBytes > 4 * M allowed in IA32 in win11 or greater even if EnableAllLargePageSizes is false

Reply:
> @nitinsingh130519 , is LargePageSizeInBytes > 4 * M allowed in IA32 in win11 or greater even if EnableAllLargePageSizes is false

No, it would default to the page size provided by the vm which for  4kb